beam/tools/src/main/java/cz/o2/proxima/beam/tools/groovy/BeamStreamProvider.java

```diff
@@ -238,32 +239,20 @@ void createUdfJarAndRegisterToPipeline(PipelineOptions opts) {
     String runnerName = opts.getRunner().getSimpleName();
     try {
       File path = createJarFromUdfs();
-      URL url = new URL("file://" + path.getAbsolutePath());
       log.info("Injecting generated jar at {} into {}", path, runnerName);
-      injectJarIntoContextClassLoader(url);
       if (opts.getRunner().getClassLoader() instanceof URLClassLoader) {
         // this is fallback
-        injectUrlIntoClassloader((URLClassLoader) opts.getRunner().getClassLoader(), url);
+        injectUrlIntoClassloader((URLClassLoader) opts.getRunner().getClassLoader(), path);
       }
-      if (runnerName.equals("SparkRunner")) {
-        List<String> filesToStage =
-            PipelineResources.detectClassPathResourcesToStage(
-                Thread.currentThread().getContextClassLoader(), opts);
-        log.info("Injecting generated jar {} into SparkRunner.filesToStage", url.toExternalForm());
+      List<String> filesToStage = Collections.singletonList(path.getAbsolutePath());
+      if (runnerName.equals("FlinkRunner")) {
+        FlinkPipelineOptions flinkOpts = opts.as(FlinkPipelineOptions.class);
+        flinkOpts.setFilesToStage(filesToStage);
+      } else if (runnerName.equals("SparkRunner")) {
         SparkCommonPipelineOptions sparkOpts = opts.as(SparkCommonPipelineOptions.class);
-        filesToStage.add(url.toExternalForm());
-        sparkOpts.setFilesToStage(
-            filesToStage
-                .stream()
-                .filter(
-                    f -> {
-                      URI uri = URI.create(f);
-                      if (!uri.isAbsolute()) {
-                        uri = URI.create("file:///" + f);
-                      }
-                      return !new File(uri).isDirectory();
-                    })
-                .collect(Collectors.toList()));
+        sparkOpts.setFilesToStage(filesToStage);
+      } else {
+        injectJarIntoContextClassLoader(path);
```

Comment:
Can we add here log message with something like "Runner not specify - probably direct"

Reply:
:+1: